src/components/FavoritesButton/FavoritesButton.tsx

```diff
@@ -1,14 +1,21 @@
+import { MouseEvent } from 'react';
 import { Heart } from 'react-feather';
 import { Button } from './FavoritesButton.styled';
 import { FavoritesButtonTestIds } from './FavoritesButtonTestIds';
 
 interface Props {
   isFavorite: boolean;
+  onClick: () => void;
 }
 
-const FavoritesButton = ({ isFavorite }: Props) => {
+const FavoritesButton = ({ isFavorite, onClick }: Props) => {
+  const handleClick = (e: MouseEvent<HTMLButtonElement>) => {
```

Comment:
**suggestion** handle favorites action inside this component.

This one will be used in many places, i think the good idea will be isolate it and handle addToFavorites action within this component